File: leet-code/numberOfSteps.go
package main

func numberOfSteps(num int) int {
	step := 0
	for num > 0 {
		if num%2 == 0 {
			num /= 2
			step++
		}
		if num%2 != 0 {
			num--
			step++
		}
	}
	return step
}

File: leet-code/numberOfSteps.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function (num) {
  let step = 0
  while (num > 0) {
    if (num % 2 === 0) {
      num /= 2
      step++
    }
    if (num % 2 !== 0) {
      num--
      step++
    }
  }
  return step
}
